src/components/raycaster.js

```diff
@@ -1,112 +1,138 @@
 var registerComponent = require('../core/component').registerComponent;
-var requestInterval = require('request-interval');
 var THREE = require('../lib/three');
 
+var scaleDummy = new THREE.Vector3();
+
+/**
+ * Raycaster component.
+ *
+ * Pass options to three.js Raycaster including which objects to test.
+ * Poll for intersections.
+ * Emit event on origin entity and on target entity on intersect.
+ *
+ * @member {array} intersectedEls - List of currently intersected entities.
+ * @member {array} objects - Cached list of meshes to intersect.
+ * @member {number} prevCheckTime - Previous time intersection was checked. To help interval.
+ * @member {object} raycaster - three.js Raycaster.
+ */
 module.exports.Component = registerComponent('raycaster', {
+  schema: {
+    far: {default: Infinity}, // Infinity.
+    interval: {default: 100},
+    near: {default: 0},
+    objects: {default: ''},
+    recursive: {default: true}
+  },
+
   init: function () {
+    this.direction = new THREE.Vector3();
+    this.directionHelper = new THREE.Quaternion();
+    this.intersectedEls = [];
+    this.objects = null;
+    this.originVec3 = new THREE.Vector3();
+    this.prevCheckTime = null;
```

Comment:
it should be ```undefined``` then. null is used when a variable that should point to an object points to nothing. In this case prevCheckTime is a number not an object so the absence of a value is ```undefined```